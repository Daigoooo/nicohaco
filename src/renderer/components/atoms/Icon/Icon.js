// @flow

import React from 'react';
import cx from 'classnames';
import photon from '../../../styles/photon';

type Props = {
  icon: string;
  size?: string;
};

const Icon = (props: Props) => (
  <span
    className={cx(photon['icon'], photon[`icon-${props.icon}`])}
    style={{
      fontSize: props.size ? props.size : '1rem'
    }}
  />
);

export default Icon;
