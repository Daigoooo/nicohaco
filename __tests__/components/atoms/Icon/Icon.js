import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Icon from '../../../../src/renderer/components/atoms/Icon';

describe('Label component', () => {
  const createDOM = (props = {}) => shallow(<Icon {...props} />);

  it('should render self and subcomponents', () => {
    const wrapper = createDOM();

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
