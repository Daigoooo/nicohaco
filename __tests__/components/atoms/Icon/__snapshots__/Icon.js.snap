// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label component should render self and subcomponents 1`] = `
<span
  className="icon icon-undefined"
  style={
    Object {
      "fontSize": "1rem",
    }
  }
/>
`;
